cpp/daal/src/data_management/roc_auc_score.cpp

```diff
@@ -0,0 +1,135 @@
+/** file roc_auc_score.cpp */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "data_management/data/internal/roc_auc_score.h"
+#include "data_management/data/numeric_table.h"
+#include "services/daal_defines.h"
+#include "src/algorithms/service_kernel_math.h"
+#include "src/algorithms/service_sort.h"
+#include "src/externals/service_dispatch.h"
+#include "src/externals/service_memory.h"
+#include "src/services/service_data_utils.h"
+#include "src/threading/threading.h"
+
+namespace daal
+{
+namespace data_management
+{
+namespace internal
+{
+template <typename FPType, daal::CpuType cpu>
+services::Status rocAucScoreImpl(const NumericTablePtr & truePrediction, const NumericTablePtr & testPrediction, FPType & score)
+{
+    services::Status s;
+    const size_t nElements = truePrediction->getNumberOfColumns();
+    ReadRows<FPType, cpu> testPredictionBlock(testPrediction.get(), 0, 1);
+    const FPType * const testPredictionPtr = testPredictionBlock.get();
+    DAAL_CHECK_BLOCK_STATUS(testPredictionBlock);
+
+    TArrayScalable<IdxValType<FPType>, cpu> predict(nElements);
+    DAAL_CHECK_MALLOC(predict.get());
+
+    TArray<FPType, cpu> predictedRank(nElements);
+    DAAL_CHECK_MALLOC(predictedRank.get());
+
+    const size_t blockSizeDefault = 256;
```

Comment:
Minnor: The work is very simple for a parallel region. I think overhead costs will take a lot of time. Trying to look at the block size experimentally.